gapic/samplegen/samplegen.py

```diff
@@ -997,18 +997,24 @@ def generate_request_object(api_schema: api.API, service: wrappers.Service, mess
     return request
 
 
-def _transport_type_from_transport(transport: str) -> str:
-    if transport == api.TRANSPORT_GRPC:
+def _sync_or_async_from_transport(transport: str) -> str:
+    if transport in (api.TRANSPORT_GRPC, api.TRANSPORT_REST):
         return "sync"
-    elif transport == api.TRANSPORT_GRPC_ASYNC:
+    else:  # transport is api.TRANSPORT_GRPC_ASYNC
+        # Currently the REST transport does not support async.
         return "async"
-    else:  # api.TRANSPORT_REST
-        return "rest"
```

Comment:
Now method will return "sync" whereas before it was "rest". Will this break anything?

Reply:
So far we have not been generating anything for the REST transport so that (incorrect) "rest" segment of the region tag has not been used.

- Generator code that skip over all REST transport snippets: https://github.com/googleapis/gapic-generator-python/blob/main/gapic/generator/generator.py#L157-L160
- Reference region tags in .NET using "sync" and "async" for REST-only Compute API:
  - sync: https://github.com/googleapis/google-cloud-dotnet/blob/main/apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.GeneratedSnippets/AcceleratorTypesClient.GetSnippet.g.cs#L19
  - async: https://github.com/googleapis/google-cloud-dotnet/blob/main/apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.GeneratedSnippets/AcceleratorTypesClient.GetAsyncSnippet.g.cs#L19